### ISSUE TITLE
fix(security): vulnerabilities found in example-carbon-accounting

### ISSUE THREAD

**Issue by zondervancalvez** [hyperledger/cacti#2062]
List of vulnerabilities found in example-carbon-accounting image during Azure Container scan.
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ZONDER~1.CAL/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ZONDER~1.CAL/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

VULNERABILITY ID | PACKAGE NAME | SEVERITY
-- | -- | --
CVE-2021-39167 | @openzeppelin/contracts | CRITICAL
CVE-2021-41264 | @openzeppelin/contracts | CRITICAL
CVE-2021-46320 | @openzeppelin/contracts | HIGH
CVE-2022-21676 | engine.io | HIGH
CVE-2021-3918 | json-schema | CRITICAL
CVE-2021-30246 | jsrsasign | CRITICAL
CVE-2022-24771 | node-forge | HIGH
CVE-2022-24772 | node-forge | HIGH
CVE-2021-23358 | underscore | HIGH

</body>

</html>

**Comment by petermetz** [comment]
P4 because this container image is not meant to be used in production.

**Comment by zondervancalvez** [comment]
@petermetz please assign this to me. Thank you

**Comment by zondervancalvez** [comment]
CVE-2021-39167, CVE-2021-41264, CVE-2021-46320 are currently fixed in our package version.
![image](https://user-images.githubusercontent.com/69126996/172538318-7f84fcef-e9d0-4050-8e84-f095bae59038.png)

**Comment by AbhinavMir** [comment]
Could we possible modify vulnerability tables to include a local ID, such as this?



local_ID | VULNERABILITY ID | PACKAGE NAME | SEVERITY
-- | -- | -- | --
1 | CVE-2021-39167 | @openzeppelin/contracts | CRITICAL

Will be easier to cross reference what isn't an issue and what is.

**Comment by aldousalvarez** [comment]
Hello @petermetz I tried to scan the latest available version for carbon accounting backend which is v1.1.3.  The result of the [scan ](https://github.com/aldousalvarez/cactus/actions/runs/5935957205/job/16095382293?pr=5) shows that some of the vulnerabilities detected in the packages in the of carbon accounting backend we have already the fixed version or greater than based on checking the latest changes in our package.json with our latest commit. So I think we just need to issue a new release for carbon accounting backend to be able to see if there are still remaining vulnerabilities with the latest changes that we have and so that we would scan the latest and updated version.

**Comment by petermetz** [comment]
> P4 because this container image is not meant to be used in production.

@aldousalvarez Gotcha, I marked it as dependent on the issuance of the new release which will then allow you to run the scan against the latest npm package with the updated dependencies.
In the meantime please work with @jagpreetsinghsasan to pick up another task.

**Comment by petermetz** [comment]
@aldousalvarez I've managed to publish v2.0.0-alpha.1 to npm for all the packages, please re-test with those!
https://www.npmjs.com/package/@hyperledger/cactus-example-carbon-accounting-backend?activeTab=versions

**Comment by aldousalvarez** [comment]
@petermetz Based on the trivy [scan ](https://github.com/aldousalvarez/cactus/actions/runs/5935957205/job/16095382293?pr=5)on version 1.1.3 the vulnerabilities has a Total of 16 (HIGH: 14, CRITICAL: 2)

After testing the version 2.0.0-alpha.1 after this [scan ](https://github.com/aldousalvarez/cactus/actions/runs/6154795836/job/16700703294)the total remaining vulnerabilities is
Total: 13 (HIGH: 11, CRITICAL: 2) 

After checking the results based on the latest scan (version 2.0.0-alpha.1) the package.json of carbon-accounting-backend
_@openzeppelin/contracts (package.json)_ is already "4.9.3", but on the scan it says that the installed version is 4.7.3

And the rest cannot be found on the package.json so it cannot be fixed or upgraded
_http-cache-semantics (package.json) 
qs (package.json)
socket.io-parser (package.json)_

**Comment by aldousalvarez** [comment]
Depends on #2718 

Waiting for that issue to be merged and then a new package version release that has the updated dependencies which uses the fixed version of http-cache-semantics

**Comment by petermetz** [comment]
@aldousalvarez FYI: `2.0.0-alpha.2` is out now, but it still has the http cache semantics issue IIRC. Stay tuned for RC.1
